pkg/discovery/util/pod_util_test.go

```diff
@@ -187,6 +187,27 @@ func TestGetReadyPods(t *testing.T) {
 	}
 }
 
+func TestMonitored(t *testing.T) {
+	pod := newPod("pod-foo")
+	pod.Annotations = map[string]string{
+		"kubernetes.io/created-by": `{"reference":{"kind":"DaemonSet","name":"daemon-set-foo"}}`,
+	}
+
+	podWithOwnerRef := pod
+	isController := true
+	podWithOwnerRef.OwnerReferences = []metav1.OwnerReference{{Kind: "DaemonSet", Name: "daemon-set-foo", Controller: &isController}}
+
+	// Without Owner reference
+	if Monitored(pod) {
+		t.Errorf("The pod in DaemonSet shouldn't be monitored")
+	}
+
+	// With Owner reference
+	if Monitored(podWithOwnerRef) {
```

Comment:
`podWithOwnerRef` has the same `Annotations` as `pod` does.  If the plan is to test the case that Pod with `OwnerReference`, then the annotations should be cleared first.